### PR TITLE
platform_filedes_on_tmpfs

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
Replaces #3250 

Ref https://github.com/cvmfs/cvmfs/issues/3192 point 2

### CODE REVIEW

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot help

**Comment by HereThereBeDragons** [comment]
@cernvm-bot unittest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

File: cvmfs/cache_posix.cc [hunk 1]
```diff
@@ -425,11 +425,19 @@ int PosixCacheManager::Rename(const char *oldpath, const char *newpath) {
...
  * buffers.
+ * 
+ * No-op if the fd is to a file that is on a tmpfs, and so already in page cache
  */
 int PosixCacheManager::Readahead(int fd) {
   unsigned char *buf[4096];
   int nbytes;
   uint64_t pos = 0;
+
+  // no-op if fd is on tmpfs
```
Comment (jblomer) [inline cvmfs/cache_posix.cc:436]: I think it's clear enough here, I'd suggest to remove the comment

**Review by jblomer — CHANGES_REQUESTED** [review]
I realize we already have the functionality of `platform_filedes_on_tmpfs` in `GetFileSystemInfo()`. So please use that one. Also, let's set a flag for "no readahead" in `PosixCacheManager::Create`, like we do for other workarounds for caches on special file systems. This avoids having to call `statfs` on every readahead attempt.

**Comment by HereThereBeDragons** [comment]
Superseded by #3262

**Pull request closed without merging**